gateways/server/resource/start.go

```diff
@@ -261,18 +262,21 @@ func FieldSelector(selectors []v1alpha1.Selector) (fields.Selector, error) {
 }
 
 // helper method to check if the object passed the user defined filters
-func passFilters(event *InformerEvent, filter *v1alpha1.ResourceFilter) error {
-	uObj := event.Obj.(*unstructured.Unstructured)
+func passFilters(event *InformerEvent, filter *v1alpha1.ResourceFilter, startTime time.Time) bool {
 	// no filters are applied.
 	if filter == nil {
-		return nil
+		return true
 	}
-	if !strings.HasPrefix(uObj.GetName(), filter.Prefix) {
-		return errors.Errorf("resource name does not match prefix. resource-name: %s, prefix: %s", uObj.GetName(), filter.Prefix)
+	uObj := event.Obj.(*unstructured.Unstructured)
+	if len(filter.Prefix) > 0 && !strings.HasPrefix(uObj.GetName(), filter.Prefix) {
+		return false
 	}
 	created := uObj.GetCreationTimestamp()
 	if !filter.CreatedBy.IsZero() && created.UTC().After(filter.CreatedBy.UTC()) {
-		return errors.Errorf("resource is created after filter time. creation-timestamp: %s, filter-creation-timestamp: %s", created.UTC().String(), filter.CreatedBy.UTC().String())
```

Comment:
It's not good to use `error` as return and log it because it's not an error.

Reply:
Although we need to print the reason why the event did not pass filter, otherwise it is a black box to the user

Reply:
added log.